### PR TITLE
Add detect-secrets tooling

### PR DESCRIPTION
**Pull request by jonesbr17**
pre-commit config is in place, as well as a secret baseline file.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# NOTE(jonesbr): Copied from https://github.ibm.com/Whitewater/whitewater-detect-secrets/blob/master/user-config/.pre-commit-config.yaml
```
Comment (rtheis) [inline .pre-commit-config.yaml:1]: This note won't help folks outside IBM.  Are there any alternatives at https://github.com/ibm/detect-secrets which I believe is public?

File: .secrets.baseline [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      "name": "TwilioKeyDetector"
+    }
+  ],
+  "results": {
+    "assets/oauth-openshift/oauth-browser-client.yaml": [
+      {
+        "hashed_secret": "d2e2ab0f407e4ee3cf2ab87d61c31b25a74085e5",
+        "is_secret": false,
+        "is_verified": false,
+        "line_number": 14,
```
Comment (rtheis) [inline .secrets.baseline:72]: Why are we placing a secret in a configmap and not in a Kubernetes secret?  I know this comment is unrelated to this PR but nonetheless, it is a security problem.

**Review by rtheis — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jonesbr17) [reply .pre-commit-config.yaml]: Updated to point to the public repo

**Review by rtheis — APPROVED** [review]
/lgtm

**Comment by rtheis** [comment]
@csrwng ptal

File: .secrets.baseline (hunk 2 above)
Comment (csrwng) [reply .secrets.baseline:72]: The configmap is used to deliver an arbitrary CR to the child cluster via the boostrap pod. I'll open an issue to switch to a secret and have the bootstrap pod apply both configmaps and secrets.

**Review by rtheis — COMMENTED** [review]
/lgtm

**Pull request merged**